get-cabal-configuration.hs

```diff
@@ -112,14 +124,30 @@ dumpPackageDescription pkgDesc cabalFile = SList [
         sourceDirs = nub (map normalise (getSourceDirectories buildInfo cabalDir))
         exts = nub (concatMap usedExtensions buildInfo)
         langs = nub (concatMap allLanguages buildInfo)
-        deps = nub (buildDepends pkgDesc)
+        thisPackage = (pkgName . package) pkgDesc
+        deps = nub (filter (\(Dependency name _) -> name /= thisPackage) (buildDepends pkgDesc))
         otherOptions = nub (filter isFlycheckUsefulOption (concatMap (hcOptions GHC) buildInfo))
 
 dumpCabalConfiguration :: String -> IO ()
 dumpCabalConfiguration cabalFile = do
-  pkgDesc <- liftM flattenPackageDescription
-                   (readPackageDescription silent cabalFile)
-  print (dumpPackageDescription pkgDesc cabalFile)
+  genericDesc <- readPackageDescription silent cabalFile
+  -- This let block is eerily like one in Cabal.Distribution.Simple.Configure
+  let enableTest t = t { testEnabled = True }
+      flaggedTests = map (second (mapTreeData enableTest))
+                     (condTestSuites genericDesc)
+      enableBenchmark bm = bm { benchmarkEnabled = True }
+      flaggedBenchmarks = map (second (mapTreeData enableBenchmark))
+                          (condBenchmarks genericDesc)
+      genericDesc' = genericDesc { condTestSuites = flaggedTests
+                                 , condBenchmarks = flaggedBenchmarks }
+#ifdef useCompilerInfo
+      buildCompilerId = unknownCompilerInfo (CompilerId buildCompilerFlavor compilerVersion) NoAbiTag
+#else
+      buildCompilerId = CompilerId buildCompilerFlavor compilerVersion
+#endif
```

Comment:
Just a minor issue, but personally I prefer to wrap any CPP-related code into a separate function so that the ugly conditionals are neatly hidden under a proper Haskell API.  Would you mind to move this piece into a separate function, i.e. `buildCompilerID` or so?